index.d.ts

```diff
@@ -3,18 +3,12 @@ import { Document, Model, Query } from 'mongoose';
 export interface ISoftDeletedDocument extends Document {
   deleted: Boolean;
   deletedAt: Date;
-};
+  softdelete: (callback: (err: any, model: ISoftDeletedDocument) => void) => void;
+  restore: (callback: (err: any, model: ISoftDeletedDocument) => void) => void;
+}
 
-export interface ISoftDeletedModel extends ISoftDeletedDocument {
-  softdelete: (
-    callback: (err: any, model: Model<ISoftDeletedDocument>) => void
-  ) => void;
+export type ISoftDeletedModel = Model<ISoftDeletedDocument>;
 
-  restore: (
-    callback: (err: any, model: Model<ISoftDeletedDocument>) => void
-  ) => void;
-};
-
-export interface ISoftDeletedQuery extends Query {
-  isDeleted: (cond: Boolean = true) => Query;
-};
\ No newline at end of file
+export interface ISoftDeletedQuery extends Query<any> {
+  isDeleted: (condition: Boolean) => Query<any>;
```

Comment:
This is still throwing errors when used in a chained query - 
`Property 'isDeleted' does not exist on type 'DocumentQuery<ITestDocument[], ITestDocument, {}>'. Did you mean 'selected'?`

Reply:
I not found any way to improve typing of this because `.find` is a mongoose method and we can't change signature. The only workaround I to manually cast with `as`:

~~~ts
(Test.find({}) as unknown as ISoftDeletedDocumentQuery).isDeleted(false)
~~~

Reply:
Got it. Something that should be added to the README so that people are aware that they need to cast this manually while using the method.

Reply:
I just add example on readme.